source/robot_model/src/Model.cpp

```diff
@@ -102,67 +124,80 @@ bool Model::init_qp_solver() {
   // set the initial data of the QP solver_
   this->solver_.data()->setNumberOfVariables(nb_joints + 1);
   this->solver_.data()->setNumberOfConstraints(this->lower_bound_constraints_.size());
-  if (!this->solver_.data()->setHessianMatrix(this->hessian_)) return false;
-  if (!this->solver_.data()->setGradient(this->gradient_)) return false;
-  if (!this->solver_.data()->setLinearConstraintsMatrix(this->constraint_matrix_)) return false;
-  if (!this->solver_.data()->setLowerBound(this->lower_bound_constraints_)) return false;
-  if (!this->solver_.data()->setUpperBound(this->upper_bound_constraints_)) return false;
+  if (!this->solver_.data()->setHessianMatrix(this->hessian_)) { return false; }
+  if (!this->solver_.data()->setGradient(this->gradient_)) { return false; }
+  if (!this->solver_.data()->setLinearConstraintsMatrix(this->constraint_matrix_)) { return false; }
+  if (!this->solver_.data()->setLowerBound(this->lower_bound_constraints_)) { return false; }
+  if (!this->solver_.data()->setUpperBound(this->upper_bound_constraints_)) { return false; }
   // instantiate the solver_
   return this->solver_.initSolver();
 }
 
-const StateRepresentation::Jacobian Model::compute_jacobian(const StateRepresentation::JointState& joint_state,
-                                                            unsigned int joint_id) {
-  using namespace pinocchio;
+StateRepresentation::Jacobian Model::compute_jacobian(const StateRepresentation::JointState& joint_state,
+                                                      int frame_id) {
+  if ((int) joint_state.get_size() != this->robot_model_.nq) {
```

Comment:
C style casts should be avoided. Also, the order of operations could be more clear. Use `static_cast<int>(joint_state.get_size())` instead.